src/main/java/com/google/devtools/build/lib/syntax/Lexer.java

```diff
@@ -809,10 +812,26 @@ private void tokenize() {
           popParen();
           break;
         case '>':
-          setToken(TokenKind.GREATER, pos - 1, pos);
+          if (lookaheadIs(0, '>') && lookaheadIs(1, '=')) {
```

Comment:
Unfortunately the tokenizeTwoChars function has set a bad precedent that makes your task harder than it ought to be. I suggest you get rid of tokenizeTwoChars and rework the tokenize function to more simply step through each state, like the Go code does at https://github.com/google/starlark-go/blob/master/syntax/scan.go#L690.

Reply:
I don't see how getting rid of tokenizeTwoChars would change this part of the code. As I can in the Go code, handling the case of 2 chars followed by an equal is done almost the same way:
https://github.com/google/starlark-go/blob/master/syntax/scan.go#L724-L734
Don't you think restructuring the tokenize function is out of the scope of this PR?

Reply:
The advantage of the Go code is that it more closely resembles the state transition chart: it only looks one character ahead. Once it has seen `>`, for example, the only three cases  are `>`, `=`, or something else. Once it sees the second `>`, the only possibilities are `>>` and `>>=`. In effect the Java implementation is backtracking more.

For lexers written in C and Go, these kinds of optimizations make a measurable difference, so it is reasonable to assume they do in Java too, even if our profiling tools won't give us statement-level CPU cost information.

But I take your point that this change follows the prevailing style used by `//=`, for example, so you needn't eliminate tokenizeTwoChars in this PR.